pinecone/core/grpc/index_grpc.py

```diff
@@ -264,37 +298,142 @@ def _vector_transform(item):
                 return GRPCVector(id=id, values=values, metadata=dict_to_proto_struct(metadata) or {})
             raise ValueError(f"Invalid vector value passed: cannot interpret type {type(item)}")
 
-        request = UpsertRequest(vectors=list(map(_vector_transform, vectors)), **kwargs)
         timeout = kwargs.pop('timeout', None)
+
+        request = UpsertRequest(vectors=list(map(_vector_transform, vectors)), **args_dict, **kwargs)
```

Comment:
I don't see why we need this `args_dict`.  
It's a different thing if we would have re-used it throughout the function many times. But we are only passing these params to a single function call.   

Why not pass the params explicitly?
```
request = UpsertRequest(vectors=list(map(_vector_transform, vectors)), namespace=namespace, **kwargs)
```
?

Reply:
The idea is to avoid passing default values if the user didn't provide something, so the server can have full control on the default params. I used the args_dict since the function creates it removes keys with None values from the dict

Reply:
I see. That's clever! 

Maybe we should change the function name to something like `parse_non_empty_args()` or `remove_none_args()` or something more explicit?

Reply:
BTW - since these are optional params, I'm pretty sure that `OpenAPI` treats `None` as "empty" and fills in the appropriate default value anyway (on the server side).  
We can do a quick check to verify, and then we maybe don't even need this mechanism.

Reply:
I liked `parse_non_empty_args` better so renamed to that